File: Tests/CSharp/Stream/Stream.cs
// **********************************************************************
//
// Copyright (c) 2009-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Test
{
    public class Stream
    {
        public static bool stream()
        {
            Ice.Communicator communicator = Ice.Util.initialize();
            Ice.OutputStream output = Ice.Util.createOutputStream(communicator);
            PersonPrx person = PersonPrxHelper.uncheckedCast(communicator.stringToProxy("person:default"));
            PersonPrxHelper.write(output, person);
            communicator.destroy();
            return true;
        }
    }
}

File: Tests/Cpp/Stream/Stream.cpp
// **********************************************************************
//
// Copyright (c) 2009-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <Ice\Ice.h>
#include <Test.h>

#include "Stream.h"

bool Test::stream()
{
	Ice::CommunicatorPtr communicator = Ice::initialize();
	Ice::OutputStreamPtr output = Ice::createOutputStream(communicator);
    Test::PersonPrx person = Test::PersonPrx::uncheckedCast(communicator->stringToProxy("person:default"));
    output->write(person);
    communicator->destroy();
	return true;
}